autopush/db.py

```diff
@@ -293,37 +297,47 @@ def create_router_table(tablename="router", read_throughput=5,
     return table
 
 
-def _drop_table(tablename):
+def _drop_table(tablename, boto_resource):
+    # type: (str, DynamoDBResource) -> None
     try:
-        g_client.delete_table(TableName=tablename)
+        boto_resource.meta.client.delete_table(TableName=tablename)
     except ClientError:  # pragma nocover
         pass
 
 
-def _make_table(table_func, tablename, read_throughput, write_throughput):
-    # type: (Callable[[str, int, int], Table], str, int, int) -> Table
+def _make_table(
+        table_func,        # type: Callable[[str, int, int, ServiceResource]]
+        tablename,         # type: str
+        read_throughput,   # type: int
+        write_throughput,  # type: int
+        boto_resource      # type: DynamoDBResource
+        ):
+    # type (...) -> DynamoDBTable
     """Private common function to make a table with a table func"""
-    if not table_exists(tablename):
-        return table_func(tablename, read_throughput, write_throughput)
+    if not boto_resource:
+        raise AutopushException("No boto3 resource provided for _make_table")
+    if not table_exists(tablename, boto_resource):
+        return table_func(tablename, read_throughput, write_throughput,
```

Comment:
This should return the table wrapped in DynamoDBTable like the clause below

Reply:
Sorry, I'm not sure I follow.

If a table doesn't exist, it must be created. We're passing the `create_router_table` function as the `table_func` param, since it contains the full table schema, which would not be available to the generic `DynamoDBTable`.

Are you suggesting that I move `create_router_table` as a class method of `DynamoDBTable` (which I could understand), but then, shouldn't I also do the same for `create_initial_message_tables`?

Reply:
What I mean is I think get_router_table and get_rotating_message_table should both return a thread safe DynamoDBTable wrapper. get_router_table/_make_table as is are returning either a DynamoDBTable wrapper if the table exists, otherwise the low level boto3 Table. That doesn't make much sense

Reply:
Ah! Ok. I agree.

Reply:
Modified `_create_*_table` to return `DynamoDBTable` instances.